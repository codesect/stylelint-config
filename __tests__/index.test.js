const fs = require('fs');
const stylelint = require('stylelint');

const config = require('../index');
const invalidCss = fs.readFileSync('./__tests__/invalid.css', 'utf-8');
const validCss = fs.readFileSync('./__tests__/valid.css', 'utf-8');

describe('Valid CSS config', () => {
  let result = null;

  beforeEach(async () => {
    result = await stylelint.lint({
      code: validCss,
      config,
    });
  });

  it('does not error if CSS is valid', () => {
    expect(result.errored).toBeFalsy();
  });

  it('raises no warnings if CSS is valid', () => {
    expect(result.results[0].warnings.length).toBe(0);
  });
});

describe('Invalid CSS config', () => {
  let result = null;

  beforeEach(async () => {
    result = await stylelint.lint({
      code: invalidCss,
      config,
    });
  });

  it('throws error if CSS is invalid', () => {
    expect(result.errored).toBeTruthy();
  });

  it('matches snapshot', () => {
    expect(result.results[0].warnings).toMatchSnapshot();
  });
});
