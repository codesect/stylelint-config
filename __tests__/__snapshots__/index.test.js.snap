// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invalid CSS config matches snapshot 1`] = `
Array [
  Object {
    "column": 3,
    "line": 198,
    "rule": "a11y/no-obsolete-attribute",
    "severity": "error",
    "text": "Unexpected using obsolete attribute \\"img[align]\\" (a11y/no-obsolete-attribute)",
  },
  Object {
    "column": 4,
    "line": 204,
    "rule": "a11y/no-obsolete-element",
    "severity": "error",
    "text": "Unexpected using obsolete selector \\"blink\\" (a11y/no-obsolete-element)",
  },
  Object {
    "column": 5,
    "line": 209,
    "rule": "a11y/no-outline-none",
    "severity": "error",
    "text": "Unexpected using \\"outline\\" property in .no-outline:focus (a11y/no-outline-none)",
  },
  Object {
    "column": 6,
    "line": 214,
    "rule": "a11y/no-text-align-justify",
    "severity": "error",
    "text": "Unexpected using \\"{ text-align: justify; }\\" in .no-justify (a11y/no-text-align-justify)",
  },
  Object {
    "column": 17,
    "line": 3,
    "rule": "at-rule-semicolon-space-before",
    "severity": "error",
    "text": "Unexpected whitespace before \\";\\" (at-rule-semicolon-space-before)",
  },
  Object {
    "column": 6,
    "line": 6,
    "rule": "at-rule-name-space-after",
    "severity": "error",
    "text": "Expected single space after at-rule name \\"@media\\" (at-rule-name-space-after)",
  },
  Object {
    "column": 1,
    "line": 13,
    "rule": "at-rule-no-vendor-prefix",
    "severity": "error",
    "text": "Unexpected vendor-prefixed at-rule \\"@-webkit-keyframes\\" (at-rule-no-vendor-prefix)",
  },
  Object {
    "column": 1,
    "line": 18,
    "rule": "comment-empty-line-before",
    "severity": "error",
    "text": "Expected empty line before comment (comment-empty-line-before)",
  },
  Object {
    "column": 4,
    "line": 23,
    "rule": "declaration-block-semicolon-newline-after",
    "severity": "error",
    "text": "Expected newline after \\";\\" in a multi-line declaration block (declaration-block-semicolon-newline-after)",
  },
  Object {
    "column": 2,
    "line": 23,
    "rule": "declaration-block-semicolon-newline-before",
    "severity": "error",
    "text": "Unexpected whitespace before \\";\\" in a multi-line declaration block (declaration-block-semicolon-newline-before)",
  },
  Object {
    "column": 2,
    "line": 23,
    "rule": "declaration-block-semicolon-space-before",
    "severity": "error",
    "text": "Unexpected whitespace before \\";\\" (declaration-block-semicolon-space-before)",
  },
  Object {
    "column": 13,
    "line": 28,
    "rule": "declaration-colon-space-after",
    "severity": "error",
    "text": "Expected single space after \\":\\" with a single-line declaration (declaration-colon-space-after)",
  },
  Object {
    "column": 14,
    "line": 33,
    "rule": "declaration-no-important",
    "severity": "error",
    "text": "Unexpected !important (declaration-no-important)",
  },
  Object {
    "column": 3,
    "line": 38,
    "rule": "declaration-property-value-disallowed-list",
    "severity": "error",
    "text": "Unexpected value \\"none\\" for property \\"border\\" (declaration-property-value-disallowed-list)",
  },
  Object {
    "column": 16,
    "line": 43,
    "rule": "font-family-name-quotes",
    "severity": "error",
    "text": "Expected quotes around \\"Source Code Pro\\" (font-family-name-quotes)",
  },
  Object {
    "column": 16,
    "line": 48,
    "rule": "font-weight-notation",
    "severity": "error",
    "text": "Expected numeric font-weight notation (font-weight-notation)",
  },
  Object {
    "column": 20,
    "line": 53,
    "rule": "function-parentheses-space-inside",
    "severity": "error",
    "text": "Unexpected whitespace after \\"(\\" in a single-line function (function-parentheses-space-inside)",
  },
  Object {
    "column": 24,
    "line": 53,
    "rule": "function-parentheses-space-inside",
    "severity": "error",
    "text": "Unexpected whitespace before \\")\\" in a single-line function (function-parentheses-space-inside)",
  },
  Object {
    "column": 25,
    "line": 58,
    "rule": "function-url-no-scheme-relative",
    "severity": "error",
    "text": "Unexpected scheme-relative url (function-url-no-scheme-relative)",
  },
  Object {
    "column": 25,
    "line": 63,
    "rule": "function-url-quotes",
    "severity": "error",
    "text": "Expected quotes (function-url-quotes)",
  },
  Object {
    "column": 9,
    "line": 72,
    "rule": "media-feature-name-no-vendor-prefix",
    "severity": "error",
    "text": "Unexpected vendor-prefix (media-feature-name-no-vendor-prefix)",
  },
  Object {
    "column": 1,
    "line": 83,
    "rule": "no-duplicate-selectors",
    "severity": "error",
    "text": "Unexpected duplicate selector \\"p\\", first used at line 27 (no-duplicate-selectors)",
  },
  Object {
    "column": 1,
    "line": 92,
    "rule": "no-duplicate-selectors",
    "severity": "error",
    "text": "Unexpected duplicate selector \\".blue\\", first used at line 88 (no-duplicate-selectors)",
  },
  Object {
    "column": 1,
    "line": 121,
    "rule": "no-duplicate-selectors",
    "severity": "error",
    "text": "Unexpected duplicate selector \\"a,p\\", first used at line 115 (no-duplicate-selectors)",
  },
  Object {
    "column": 1,
    "line": 124,
    "rule": "no-duplicate-selectors",
    "severity": "error",
    "text": "Unexpected duplicate selector \\"a ,p\\", first used at line 115 (no-duplicate-selectors)",
  },
  Object {
    "column": 1,
    "line": 177,
    "rule": "no-duplicate-selectors",
    "severity": "error",
    "text": "Unexpected duplicate selector \\"a\\", first used at line 21 (no-duplicate-selectors)",
  },
  Object {
    "column": 1,
    "line": 1,
    "rule": "no-empty-first-line",
    "severity": "error",
    "text": "Unexpected empty line (no-empty-first-line)",
  },
  Object {
    "column": 3,
    "line": 98,
    "rule": "property-no-vendor-prefix",
    "severity": "error",
    "text": "Unexpected vendor-prefix \\"-webkit-transform\\" (property-no-vendor-prefix)",
  },
  Object {
    "column": 1,
    "line": 105,
    "rule": "rule-empty-line-before",
    "severity": "error",
    "text": "Expected empty line before rule (rule-empty-line-before)",
  },
  Object {
    "column": 9,
    "line": 110,
    "rule": "selector-attribute-quotes",
    "severity": "error",
    "text": "Expected quotes around \\"_blank\\" (selector-attribute-quotes)",
  },
  Object {
    "column": 1,
    "line": 116,
    "rule": "selector-list-comma-newline-after",
    "severity": "error",
    "text": "Expected newline after \\",\\" (selector-list-comma-newline-after)",
  },
  Object {
    "column": 2,
    "line": 121,
    "rule": "selector-list-comma-newline-after",
    "severity": "error",
    "text": "Expected newline after \\",\\" (selector-list-comma-newline-after)",
  },
  Object {
    "column": 3,
    "line": 124,
    "rule": "selector-list-comma-newline-after",
    "severity": "error",
    "text": "Expected newline after \\",\\" (selector-list-comma-newline-after)",
  },
  Object {
    "column": 1,
    "line": 116,
    "rule": "selector-list-comma-newline-before",
    "severity": "error",
    "text": "Unexpected whitespace before \\",\\" in a multi-line list (selector-list-comma-newline-before)",
  },
  Object {
    "column": 2,
    "line": 121,
    "rule": "selector-list-comma-space-after",
    "severity": "error",
    "text": "Expected single space after \\",\\" in a single-line list (selector-list-comma-space-after)",
  },
  Object {
    "column": 3,
    "line": 124,
    "rule": "selector-list-comma-space-after",
    "severity": "error",
    "text": "Expected single space after \\",\\" in a single-line list (selector-list-comma-space-after)",
  },
  Object {
    "column": 3,
    "line": 124,
    "rule": "selector-list-comma-space-before",
    "severity": "error",
    "text": "Unexpected whitespace before \\",\\" in a single-line list (selector-list-comma-space-before)",
  },
  Object {
    "column": 1,
    "line": 127,
    "rule": "selector-max-attribute",
    "severity": "error",
    "text": "Expected \\"[type=\\"number\\"][name=\\"quality\\"][disabled]\\" to have no more than 2 attribute selectors (selector-max-attribute)",
  },
  Object {
    "column": 1,
    "line": 132,
    "rule": "selector-max-class",
    "severity": "error",
    "text": "Expected \\".a .b .c .d\\" to have no more than 3 classes (selector-max-class)",
  },
  Object {
    "column": 1,
    "line": 142,
    "rule": "selector-max-class",
    "severity": "error",
    "text": "Expected \\".foo .bar .baz .selector\\" to have no more than 3 classes (selector-max-class)",
  },
  Object {
    "column": 1,
    "line": 137,
    "rule": "selector-max-combinators",
    "severity": "error",
    "text": "Expected \\"nav > ul ~ p + a i\\" to have no more than 3 combinators (selector-max-combinators)",
  },
  Object {
    "column": 1,
    "line": 132,
    "rule": "selector-max-compound-selectors",
    "severity": "error",
    "text": "Expected \\".a .b .c .d\\" to have no more than 3 compound selectors (selector-max-compound-selectors)",
  },
  Object {
    "column": 1,
    "line": 137,
    "rule": "selector-max-compound-selectors",
    "severity": "error",
    "text": "Expected \\"nav > ul ~ p + a i\\" to have no more than 3 compound selectors (selector-max-compound-selectors)",
  },
  Object {
    "column": 1,
    "line": 142,
    "rule": "selector-max-compound-selectors",
    "severity": "error",
    "text": "Expected \\".foo .bar .baz .selector\\" to have no more than 3 compound selectors (selector-max-compound-selectors)",
  },
  Object {
    "column": 1,
    "line": 147,
    "rule": "selector-max-id",
    "severity": "error",
    "text": "Expected \\"#id\\" to have no more than 0 ID selectors (selector-max-id)",
  },
  Object {
    "column": 1,
    "line": 137,
    "rule": "selector-max-type",
    "severity": "error",
    "text": "Expected \\"nav > ul ~ p + a i\\" to have no more than 2 type selectors (selector-max-type)",
  },
  Object {
    "column": 1,
    "line": 152,
    "rule": "selector-max-type",
    "severity": "error",
    "text": "Expected \\"div a span\\" to have no more than 2 type selectors (selector-max-type)",
  },
  Object {
    "column": 1,
    "line": 157,
    "rule": "selector-max-universal",
    "severity": "error",
    "text": "Expected \\"* *\\" to have no more than 1 universal selector (selector-max-universal)",
  },
  Object {
    "column": 1,
    "line": 162,
    "rule": "selector-no-qualifying-type",
    "severity": "error",
    "text": "Unexpected qualifying type selector (selector-no-qualifying-type)",
  },
  Object {
    "column": 7,
    "line": 167,
    "rule": "selector-no-vendor-prefix",
    "severity": "error",
    "text": "Unexpected vendor-prefix \\"::-moz-placeholder\\" (selector-no-vendor-prefix)",
  },
  Object {
    "column": 3,
    "line": 173,
    "rule": "shorthand-property-no-redundant-values",
    "severity": "error",
    "text": "Unexpected longhand value '1rem 1rem 1rem 1rem' instead of '1rem' (shorthand-property-no-redundant-values)",
  },
  Object {
    "column": 12,
    "line": 178,
    "rule": "string-quotes",
    "severity": "error",
    "text": "Expected double quotes (string-quotes)",
  },
  Object {
    "column": 12,
    "line": 183,
    "rule": "value-keyword-case",
    "severity": "error",
    "text": "Expected \\"Block\\" to be \\"block\\" (value-keyword-case)",
  },
  Object {
    "column": 21,
    "line": 188,
    "rule": "value-list-comma-space-after",
    "severity": "error",
    "text": "Expected single space after \\",\\" in a single-line list (value-list-comma-space-after)",
  },
  Object {
    "column": 12,
    "line": 193,
    "rule": "value-no-vendor-prefix",
    "severity": "error",
    "text": "Unexpected vendor-prefix \\"-webkit-flex\\" (value-no-vendor-prefix)",
  },
  Object {
    "column": 5,
    "line": 68,
    "rule": "indentation",
    "severity": "error",
    "text": "Expected indentation of 2 spaces (indentation)",
  },
]
`;
